Bottom: e5d2faf273bb5e241d48185f994338b93c1b1a76
Top:    378432a42897fbab2844f38f5f82dcbe9d0d02bc
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 19:03:18 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown-2.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 96a55b3..d9d6d44 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1285,6 +1285,13 @@ static int do_hup_signal(struct master *master, time_t age)
 	nfs_mount_uses_string_options = check_nfs_mount_version(&vers, &check);
 
 	master_mutex_lock();
+	/* Already shutdown or no mounts ? */
+	if (list_empty(&master->mounts)) {
+		status = pthread_mutex_unlock(&mrc.mutex);
+		if (status)
+			fatal(status);
+		return;
+	}
 	if (master->reading) {
 		status = pthread_mutex_unlock(&mrc.mutex);
 		if (status)
@@ -1341,6 +1348,7 @@ static void *statemachine(void *arg)
 		case SIGTERM:
 		case SIGINT:
 		case SIGUSR2:
+			logerr("got signal %d", sig);
 			master_mutex_lock();
 			if (list_empty(&master_list->completed)) {
 				if (list_empty(&master_list->mounts)) {
